### PR TITLE
Update our repository to incorporate project health best practices from clomonitor.io

### LINKED ISSUE #108051: Make CloMonitor happy
See https://clomonitor.io/projects/kubernetes/kubernetes Most of the things that the tool is complaining about, we have information just not in the form the tool accepts. So let's try to see if we can point to existing resources with a placeholder file or something?

image

For example the Roadmap may be a pointer to kubernetes/enhancements repository, governance may be a pointer to relevant docs in community and/or steering repository etc.

### PR DESCRIPTION
**Pull request by whtssub**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #108051 

#### Special notes for your reviewer:
A Fossa badge under the License section of the README.md file in k/k needs to be added to completely fix this issue. 
#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
/retitle Update our repository to incorporate project health best practices from clomonitor.io

File: README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Kubernetes (K8s)
 
-[![GoPkg Widget]][GoPkg] [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/569/badge)](https://bestpractices.coreinfrastructure.org/projects/569)
+[![GoPkg Widget]][GoPkg] [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/569/badge)](https://bestpractices.coreinfrastructure.org/projects/569) [![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fkubernetes%2Fkubernetes.svg?type=shield)](https://app.fossa.io/projects/git%2Bgithub.com%2Fkubernetes%2Fkubernetes?ref=badge_shield)
```
Comment (liggitt) [inline README.md:3]: what are the implications of embedding an image from another domain? does that expose people visiting this repo to tracking/metrics from that site?

**Comment by liggitt** [comment]
I'm not very familiar with clomonitor.io ... do we think it's reasonable to require all OSS projects to add this many root dir files and a badge to the readme to get a good score?

File: README.md (hunk 1 above)
Comment (dims) [reply README.md:3]: good point @liggitt 

Also looks like we are not ready to do this yet? see slack conversation with Ihor etc at:
https://kubernetes.slack.com/archives/C01672LSZL0/p1644869015702139
https://github.com/cncf/clomonitor/issues/50

@SubhasmitaSw please more the FOSSA Status for now.

**Comment by liggitt** [comment]
(@SubhasmitaSw thanks for opening the PR, it looks mechanically correct ... my questions are more for the issue author @dims and the clomonitor folks like @idvoretskyi)

File: README.md (hunk 1 above)
Comment (SubhasmitaSw) [reply README.md:3]: > I'm not very familiar with clomonitor.io ... do we think it's reasonable to require all OSS projects to add this many root dir files and a badge to the readme to get a good score?

The typical trend that I observed from other 100% scored projects on cloMonitor dashboard had these implementations in common.

Comment (liggitt) [reply README.md:3]: answering my own question... for the privacy bit at least, it looks like github scrapes it and serves it through a `https://camo.githubusercontent.com/...` domain

**Comment by dims** [comment]
> I'm not very familiar with clomonitor.io ... do we think it's reasonable to require all OSS projects to add this many root dir files and a badge to the readme to get a good score?

This clomonitor is a new tool (currently in Alpha) where CNCF is trying to set some best practices across projects. This is our way of giving feedback. Take some of the things they are suggesting as it make sense and push back on others.

**Comment by dims** [comment]
The main reason for the FOSSA one is apparently to ensure license scanning is automated in a repo. So i replied back that we are using Snyk and we post to our testgrid.

**Comment by parispittman** [comment]
/lgtm

appreciate the general summaries with links back to the original sources so the material won't need to change in many places. tysm!!

**Comment by dims** [comment]
@liggitt ok with you? looks like Paris is ok with the rest of the things here.

**Review by liggitt — COMMENTED** [review]
noted a few typos / tweaks

Apart from that, I'm not thrilled with littering the root dir with a bunch of files to make a scanner happy, and I'm not sure this actually makes this content more discoverable to a user than just putting it in the readme.

https://github.com/cncf/clomonitor/issues/67 already referenced this PR as an example of Kubernetes conforming to a "standard" location for these files, but are these files actually standard?

It seems like specific headers / sections in the readme.md would be equally discoverable by the scanner, less clutter in the root dir, and more discoverable by users (since github renders the content when viewing the repo by default). Could we propose the scanner recognize headers like this in the readme in addition to the standalone files?

```
## Roadmap

...

## Governance

...

## Adopters

...
```

**Comment by SubhasmitaSw** [comment]
> noted a few typos / tweaks
> 
> Apart from that, I'm not thrilled with littering the root dir with a bunch of files to make a scanner happy, and I'm not sure this actually makes this content more discoverable to a user than just putting it in the readme.
> 
> [cncf/clomonitor#67](https://github.com/cncf/clomonitor/issues/67) already referenced this PR as an example of Kubernetes conforming to a "standard" location for these files, but are these files actually standard?
> 
> It seems like specific headers / sections in the readme.md would be equally discoverable by the scanner, less clutter in the root dir, and more discoverable by users (since github renders the content when viewing the repo by default). Could we propose the scanner recognize headers like this in the readme in addition to the standalone files?
> 
> ```
> ## Roadmap
> 
> ...
> 
> ## Governance
> 
> ...
> 
> ## Adopters
> 
> ...
> ```

@dims @liggitt So, shall we proceed with the interim adjustments for the time being? till cloMonitor provides this support?

**Comment by tegioz** [comment]
Hi! 👋 

This is Sergio, working on CloMonitor 🙂 

Detecting those headers in the `README` file for some checks sounds like a good alternative, so we'd be happy to add support for that. We are actually doing something similar for some checks, like *Community meetings*. So if you feel more comfortable this way than adding those extra files we can take care of it soon.

**Comment by tegioz** [comment]
It's ready, we now detect markdown headers (#+ format) in the `README` file for the following checks:

- Adopters
- Code of conduct
- Governance
- Roadmap
- Security policy

The regular expressions are not too rigid (sample: `(?im)^#+.*adopters.*$`), so you should be able to use headers like `# Kubernetes adopters` or `## Adopters`.

If you run into any issue please let us know. Thanks!

**Comment by dims** [comment]
@tegioz thanks!

@SubhasmitaSw looks like  CloMonitor updates are ready. So let's go ahead with the new markdown headers that they are looking for.

**Comment by SubhasmitaSw** [comment]
> @tegioz thanks!
> 
> @SubhasmitaSw looks like CloMonitor updates are ready. So let's go ahead with the new markdown headers that they are looking for.

I've made the changes just pushing it out.

**Pull request merged**